alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -275,19 +292,29 @@ private function startConcat($batchJob)
 		$tempEntry = entryPeer::retrieveByPK($jobData->getTempEntryId());
 		$assets = assetPeer::retrieveByEntryId($jobData->getTempEntryId(), array(assetType::FLAVOR));
 		usort($assets, array("kClipManager","cmpByOrder"));
-
+		
 		$files = $this->getFilesPath($assets);
-
-		$flavorAsset = $this->addNewAssetToTargetEntry($tempEntry);
-
-		//calling addConcatJob only if lock succeeds
-		$store = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_LOCK_KEYS);
-		$lockKey = "kclipManager_add_concat_job" . $batchJob->getId();
-		if (!$store || $store->add($lockKey, true, self::LOCK_EXPIRY))
+		$sourceAssetIds = array_keys($files);
+		foreach ($sourceAssetIds as $assetId)
```

Comment:
instead if calling array_keys($files); and $this->filterByAssetId($files, $assetId);
all you need is -
foreach($files as $assetId => $relatedFiles)
{
.....